geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/IndexRepositoryFactory.java

```diff
@@ -50,19 +50,29 @@
   public IndexRepositoryFactory() {}
 
   public IndexRepository computeIndexRepository(final Integer bucketId, LuceneSerializer serializer,
-      LuceneIndexImpl index, PartitionedRegion userRegion, final IndexRepository oldRepository)
-      throws IOException {
+      LuceneIndexImpl index, PartitionedRegion userRegion, final IndexRepository oldRepository,
+      final AbstractPartitionedRepositoryManager partitionedRepositoryManager) throws IOException {
     LuceneIndexForPartitionedRegion indexForPR = (LuceneIndexForPartitionedRegion) index;
     final PartitionedRegion fileRegion = indexForPR.getFileAndChunkRegion();
-
     // We need to ensure that all members have created the fileAndChunk region before continuing
     Region prRoot = PartitionedRegionHelper.getPRRoot(fileRegion.getCache());
     PartitionRegionConfig prConfig =
         (PartitionRegionConfig) prRoot.get(fileRegion.getRegionIdentifier());
-    while (!prConfig.isColocationComplete()) {
-      prConfig = (PartitionRegionConfig) prRoot.get(fileRegion.getRegionIdentifier());
+    LuceneFileRegionColocationCompleteListener luceneFileRegionColocationCompleteListener =
+        new LuceneFileRegionColocationCompleteListener(partitionedRepositoryManager, bucketId);
+    fileRegion.setColocationCompleteListener(luceneFileRegionColocationCompleteListener);
+    IndexRepository repo = null;
+    if (prConfig.isColocationComplete()) {
+      repo = getIndexRepository(bucketId, serializer, userRegion, oldRepository, index);
     }
+    return repo;
+  }
 
+  private IndexRepository getIndexRepository(Integer bucketId, LuceneSerializer serializer,
```

Comment:
I'm torn on this (and may have even suggested extracting this out originally) but looking at it now, I'd want to prevent anyone else from accidentally calling this method outside of computeRepository.  Would we want to stick this into compute repository itself at this point or comment that this should only be called from compute repository?

At the least, we should rename this method from getIndexRepository.  Maybe finishComputingRepository or createRepository.  getRepository seems like it would unintentionally be called to just do a simple getter.